docs/user-guide/rug/generators.md

```diff
@@ -0,0 +1,453 @@
+import { PopulateProject } from "@atomist/rug/operations/ProjectGenerator"
+import { Generator, Tags } from '@atomist/rug/operations/Decorators'
+import { Pattern } from '@atomist/rug/operations/RugOperations'
+import { Project } from '@atomist/rug/model/Project'
+
+@Generator("NewMkDocsDocumentationProject", "creates a MkDocs project")
+@Tags("markdown", "documentation", "mkdocs")
+class NewMkDocsDocumentationProject implements PopulateProject {
+
+    @Parameter({description: "name of the default page", pattern: Pattern.any})
+    defaultdoc: string = "index.md"
+
+    populate(project: Project) {
+        // set the created project's name in the original source
+        project.findFile("mkdocs.yml").replace("<NAME>", project.name())
+
+        // create the docs directory
+        project.addDirectory("docs")
+
+        // call an editor to add a default document
+        project.editWith("AddMarkdownDocumentToMkDocsProject", {
+            "name": "docs/" + this.defaultdoc,
+            "title": "My new project",
+            "content": "Hi there!"
+        })
+    }
+}
+export let generator = new NewMkDocsDocumentationProject()
+```
+
+Please see below for a [detailed description](#what-is-a-generator-made-of)
+of this generator's structure.
+
+### What do Rug Generators do?
+
+A generator does the followings:
+
+1. Copy the content of the [Rug project][rugproj] into a target directory
+2. Applies the generator's function against the filled target directory to
+   tune its content
+
+Let's assume the following Rug generator project:
+
+```console
+mkdocs-generator
+    ├── .atomist
+    │   ├── manifest.yml
+    │   ├── package.json
+    │   ├── editors
+    │   │   └── NewMkDocsDocumentationProject.ts
+    │   └── tests
+    │       └── NewMkDocsDocumentationProject.rt
+    ├── mkdocs.yml
+    ├── README.md
+    └── requirements.txt
+```
+
+That project itself is a valid [MkDocs][mkdocs] (a documentation builder tool)
+project that can be run in its own right.
+
+To create a project from this Rug generator, run the following [rug][cli] 
+command:
+
+```console
+$ cd ~/workspace
+$ rug generate NewMkDocsDocumentationProject my-project-docs
+```
+
+This will result in the following directory structure:
+
+```console
+my-project-docs
+    ├── docs
+    │   └── index.md
+    ├── mkdocs.yml
+    ├── README.md
+    └── requirements.txt
+```
+
+The generator copied the whole content of its source and then added an
+`docs/index.md` file as per its `populate(project)` function. It also amended
+the file `mkdocs.yml` with the generated project's name.
+
+## Search for Generators
+
+### Using the Atomist bot
+
+To see the list of available generators to your team, ask the bot for them:
+
+```
+@atomist generators
+```
+
+This will return a short list of generators, you can refine your search by
+specifying filter words to the command. For instance, to search for generators
+for Java and Spring:
+
+```
+@atomist java spring
+```
+
+### Using the CLI
```

Comment:
It's more likely the case that I hit the compose key followed by space. My editor should warn me about them indeed.